### PR TITLE
fix(watcher): keep team fallback orchestration alive while live worker panes remain

### LINKED ISSUE #1042: bug(watcher): keep fallback team orchestration alive while live worker panes remain after parent exit
## Summary
The notify fallback watcher can still stop the leader-side orchestration after the leader parent exits even when a plain team run still has live worker panes.

## Why this matters
When this happens, tmux workers keep running, but the leader control plane stops pumping fallback dispatch, leader nudges, and other wake-up behavior. From the user's perspective, the main orchestration looks stuck even though workers are still alive.

## Reproduction shape / evidence
- Start a team run where worker panes are alive in tmux.
- Let the leader parent process exit while the team remains active.
- The watcher currently shuts down because `enforceLifecycleGuards()` only defers parent-loss shutdown for active Ralph state.
- That leaves live workers without the fallback watcher control plane.

Relevant codepaths:
- `src/scripts/notify-fallback-watcher.ts`
- `src/scripts/notify-hook/team-leader-nudge.ts`

## Overlap assessment
**Classification:** Follow-up.

Related prior work:
- #1010 / PR #1011 fixed the linked `team + ralph` orchestration gap.
- PR #965 restored watcher authority for launch-path fallback handling.
- This issue is still distinct because the remaining gap affects plain active team runs when the watcher sees parent loss while worker panes are still alive.

In short:
- previous fixes covered linked-Ralph survival and authority restoration,
- this gap remains in the parent-loss lifecycle guard for active teams with live workers.

## Proposed direction
Keep the fallback watcher alive after parent loss when an active team still has live worker panes, and continue shutting it down once the team becomes terminal.

## Acceptance criteria
- The fallback watcher does not exit on parent loss while an active team still has live worker panes.
- The watcher still exits once the team becomes terminal.
- Regression coverage proves the parent-loss defer path for active teams.
- The fix does not keep the watcher alive for stale teams that have no live worker panes.

## Out of scope
- Broad redesign of max-lifetime behavior.
- General team control-plane redesign.
- Changes to unrelated tmux injection cooldown behavior.

### PR DESCRIPTION
**Pull request by HaD0Yun**
## Summary
This PR keeps the fallback watcher alive after leader parent loss when an active team still has live worker panes.

Closes #1042.

## Problem statement
Users can end up with tmux workers that are still running while the main leader-side orchestration stops. The fallback watcher is responsible for control-plane recovery work such as dispatch draining and leader nudges, but its parent-loss guard only deferred shutdown for active Ralph state.

That meant a plain team run could still lose its fallback control plane even though workers were visibly alive.

## Root cause
`enforceLifecycleGuards()` in `src/scripts/notify-fallback-watcher.ts` treated parent loss as survivable only for active Ralph state. It did not check whether an active team still had live worker panes, so the watcher exited and left the team without leader-side fallback orchestration.

## What changed
- added an active-team resolver in `src/scripts/notify-fallback-watcher.ts`
- taught the parent-loss lifecycle guard to defer shutdown when an active team still has live worker panes
- added a regression test proving the watcher survives parent loss until the active team becomes terminal

## Why this design
I kept the patch narrow and tied to the observed regression:
- it only changes the parent-loss guard
- it requires both active team state and live worker panes
- it avoids keeping the watcher alive for stale/crashed teams that only left behind state files

A broader redesign of watcher lifetime or max-lifetime handling would be riskier and belongs in a separate follow-up.

## Overlap assessment
**Classification:** Follow-up.

Related prior work:
- #1042 tracks this remaining parent-loss gap for plain active team runs
- #1010 / PR #1011 fixed linked `team + ralph` orchestration survival
- PR #965 restored launch-path watcher authority

This PR closes a remaining gap for plain active team runs: parent-loss shutdown still ignored active teams with live worker panes.

## Validation
- `npm run build`
- `npx biome lint src/scripts/notify-fallback-watcher.ts src/hooks/__tests__/notify-fallback-watcher.test.ts`
- `node --test dist/hooks/__tests__/notify-fallback-watcher.test.js`

## Risk / compatibility
- Expected behavior change: the watcher now survives parent loss for active teams that still have live worker panes.
- Unchanged behavior: the watcher still exits when the team becomes terminal, and it still shuts down for stale teams without live worker panes.
- Remaining risk: max-lifetime behavior for very long-lived active teams is unchanged in this patch.

## Files changed
- `src/scripts/notify-fallback-watcher.ts`
- `src/hooks/__tests__/notify-fallback-watcher.test.ts`

### CODE REVIEW

File: src/scripts/notify-fallback-watcher.ts [hunk 1]
```diff
@@ -348,6 +360,70 @@ async function resolveActiveRalphState(): Promise<ActiveModeResult> {
...
+    const path = join(dir, 'team-state.json');
+    if (!existsSync(path)) continue;
+    const parsed = await readFile(path, 'utf-8')
+      .then((content) => JSON.parse(content) as Record<string, unknown>)
+      .catch(() => null);
+    if (!parsed || typeof parsed !== 'object' || parsed.active !== true) continue;
+
+    const teamName = safeString(parsed.team_name).trim();
+    if (!teamName) continue;
+
```
[the remark this answers is not in the document]
Comment (HaD0Yun) [reply src/scripts/notify-fallback-watcher.ts:388]: Good catch. I updated the watcher so the active-team defer path now consults the canonical `team/<name>/phase.json` before treating a team as live. If the phase is already terminal, the watcher no longer defers parent-loss shutdown just because root `team-state.json` is still marked active.

I also added a regression test for that exact edge case alongside the existing live-worker defer test.

Validation rerun:
- `npm run build`
- `npx biome lint src/scripts/notify-fallback-watcher.ts src/hooks/__tests__/notify-fallback-watcher.test.ts`
- `node --test dist/hooks/__tests__/notify-fallback-watcher.test.js`
